Hrm04 module linking UI fixes

akshaya86 opened the pull request:
- module linking
- ui fixes
- digital watch face service added

EntyR commented:
Please add snap helper to center central element

EntyR commented:
Also please compare ui design with one from task document. It must look similar

EntyR commented:
Design template doesn't categories it's properly so I'll put it here

1st template its normal state and the rest 2 it's pressed state.

Also please watch for spacing between elements and for their size to fit design.

Spacing is a little bit big and elements are small

![image](https://user-images.githubusercontent.com/89445868/171602464-86d9fcec-4456-45ba-9f3e-2bf7d726a92e.png)

EntyR commented:
Please lighten up the central element. Can change transparency of non-central for example.

Check for elements color and font it should fit those in design template 

Also look for elements size. Only 3 element should be on the screen at the time. 

![image](https://user-images.githubusercontent.com/89445868/171602704-c8fa4bab-61cd-4b16-9fd7-a29c25d0fc38.png)

akshaya86 committed:
HRM04: Removed Unused file . Linked module

akshaya86 committed:
HRM04: Review comment fixed

akshaya86 committed:
Merge branch 'develop' of https://github.com/akshaya86/WearOSApplication into HRM04-Module-Linking-ui-fixes

 Conflicts:
	app/src/main/AndroidManifest.xml
	data/src/main/java/com/example/data/database/Converters.kt
	data/src/main/java/com/example/data/database/dao/HeartDataDao.kt
	data/src/main/java/com/example/data/database/entity/HeartDataEntity.kt
	data/src/main/java/com/example/data/datamodels/base/DatabaseResult.kt
	data/src/main/java/com/example/data/datamodels/model/HeartRate.kt
	data/src/main/java/com/example/data/di/RepositoryModule.kt
	data/src/main/java/com/example/data/repository/BaseRepository.kt
	data/src/main/java/com/example/data/repository/HeartRateMeasureRepository.kt
	domain/src/main/java/com/example/domain/base/IBaseUseCase.kt
	domain/src/main/java/com/example/domain/di/InteractionModule.kt
	domain/src/main/java/com/example/domain/model/DbResultHandler.kt
	domain/src/main/java/com/example/domain/repository/IHeartRateMeasureRepository.kt
	domain/src/main/java/com/example/domain/usecase/GetHeartRateUseCase.kt

akshaya86 commented:
@EntyR Please revisit this PR

EntyR commented:
It's not building now. Check IGetHeartRateUseCase  interface

EntyR commented:
![image](https://user-images.githubusercontent.com/89445868/172127000-566189dd-704f-47ed-a7b2-29c521bad1ba.png)
Please update spacing to match figma template, also check for pressed and not central elements design it's differ

EntyR commented:
![image](https://user-images.githubusercontent.com/89445868/172130276-d29b659d-e8e6-4f3a-9412-8ed59fbe988d.png)
Please also look for top and bottom elements:

1. It shouldn't clip any of them like in screenshot
2. you should be able to center 1'st element

EntyR commented:
![image](https://user-images.githubusercontent.com/89445868/172132563-17dde0c3-9120-48b6-9856-fb1163c4ae10.png)
Text is clipping

EntyR commented:
Look if you can use [different dimen resources](https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes) or calculate size with screen height and width for devices with bigger screen size

![image](https://user-images.githubusercontent.com/89445868/172138808-b7fcd053-f3ca-4976-87b1-2f780dd75616.png)

EntyR commented:
Please move watch face and others screen to appropriate requests if you not going to implement them here, 
also tell me if you are going to do this here, I'll test them too then

akshaya86 commented:
Hi @EntyR i will cover all UI fixes and changes in this PR only. 
Working on latest review comments.

akshaya86 committed:
HRM04: Resolved Review comments

akshaya86 commented:
@EntyR please revisit this PR.

I have tested watch screen of WearOS Large Round 30API with resoultion 454*454

EntyR commented:
![image](https://user-images.githubusercontent.com/89445868/172617106-f7eb3145-92d2-4071-afd4-2a061ccc95af.png)
It seems like where no data loaded for graph so can't test this one, still please fix positioning for element to fit figma as I descripted in the issue.

About sizing you always should use either  different [dimen resource](https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes) files for other screen size or calculate size programmatically in activity/fragment instead based on your device screen size, basically your design and design from figma should fit as much as possible and different screen sizes kind of ruin this idea, it's not an issue in this project but still nice thing to have

EntyR commented:
As for main screen please add margin for top element so it can be centered properly

akshaya86 committed:
HRM04: Resolved Review comments

akshaya86 committed:
HRM04: Screen UI fixes done for measure HR

akshaya86 committed:
HRM04: Screen UI fixes done for Export

akshaya86 committed:
HRM04: Screen UI fixes done for Export

akshaya86 committed:
HRM04: Screen UI fixes done for Export

akshaya86 commented:
@EntyR Please revisit this PR

EntyR commented:
![image](https://user-images.githubusercontent.com/89445868/172862286-7649b36c-7ecb-4abc-8a37-bf3b053b468d.png)
Still unable to center first and last element it should look like this. Try to update margin in adapter onbind viewholder method something like this 


```
val layoutParams = holder.binding.root.layoutParams as ViewGroup.MarginLayoutParams
if (currentFeature.id == 0) {
            layoutParams.updateMargins(top =myoffset)
        }
```

EntyR commented:
Testing on 320x320 resolution device

![image](https://user-images.githubusercontent.com/89445868/172864871-19f31bfd-e12c-4b44-8907-e231e0425bbc.png)

items too small 4 elements could fit on the screen and where should be only 3

EntyR commented:
Also please reduce time of waiting for export data 1 second is enough we just need to show what the screen exist and not force users to actually wait

EntyR commented:
Very nice Measure screen, but still toast messages not looking too great 

![image](https://user-images.githubusercontent.com/89445868/172866858-d06eb8e3-2b8f-4567-8fa8-9d8b01edb317.png)

Please replace with something like dialog box or just change text in start measure textview same with toast for other screen.

akshaya86 commented:
@EntyR  Thank you for revisiting . i will retest screen with 320*320 , update the PR

akshaya86 committed:
HRM04: Resolved review comments

akshaya86 commented:
@EntyR 

- Fixed all review comment also tested on emulator having resolution 384*384 as 382 is not available .
- Fixed margin issue and kept each item in centered position .
- reduced delay time 
- Removed all toast and added text error messages for respective screens .

akshaya86 committed:
HRM04: Watch face added

EntyR commented on app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt:
```diff
@@ -4,41 +4,25 @@ import android.content.BroadcastReceiver
+class MyDigitalWatchFace : CanvasWatchFaceService() {
 
     companion object {
         private val NORMAL_TYPEFACE = Typeface.create(Typeface.SANS_SERIF, Typeface.NORMAL)
```
Please use roboco font like in template

akshaya86 replied:
Resolved

EntyR commented on app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt:
```diff
@@ -175,40 +160,53 @@ class MyWatchFaceDigital : CanvasWatchFaceService() {
             invalidate()
         }
 
         override fun onDraw(canvas: Canvas, bounds: Rect) {
```
Please don't create any instances inside on draw or if you do make sure it's created only once, onDraw is called very often so it's bad practice to anything except drawing

akshaya86 replied:
Not getting comment. can you please beat explain ? regarding background & time set ?

EntyR commented:
Also for your watchface background you can use gradient instead of loading image resources, something like this

`private val colors = intArrayOf( Color.BLACK, Color.BLACK, context.getColor(R.color.purple_primary), Color.BLACK, )`
`val backgroundColor = SweepGradient( bounds.exactCenterX(), bounds.exactCenterY(), colors, null )`

`val matrix = Matrix()`
`matrix.preRotate(27f, bounds.exactCenterX(), bounds.exactCenterY()) backgroundColor.setLocalMatrix(matrix) mainPaint.shader = backgroundColor`

akshaya86 committed:
HRM04: Font added from assets

akshaya86 committed:
Merge branch 'develop' of https://github.com/akshaya86/WearOSApplication into HRM04-Module-Linking-ui-fixes

 Conflicts:
	app/build.gradle
	app/src/main/AndroidManifest.xml
	app/src/main/java/com/example/wearableapp/di/ViewModelModule.kt
	app/src/main/java/com/example/wearableapp/presentation/adapter/MainMenuAdapter.kt
	app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt
	app/src/main/java/com/example/wearableapp/presentation/ui/MainActivity.kt
	app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt
	app/src/main/java/com/example/wearableapp/presentation/viewmodel/ExportHeartRateViewModel.kt
	app/src/main/java/com/example/wearableapp/presentation/viewmodel/MainViewModel.kt
	app/src/main/java/com/example/wearableapp/presentation/viewmodel/MeasureHeartRateViewModel.kt
	app/src/main/res/drawable-nodpi/rounded_bg.xml
	app/src/main/res/drawable/circular_progress_indicator.xml
	app/src/main/res/font/roboto_regular.ttf
	app/src/main/res/layout/activity_export_heart_rate.xml
	app/src/main/res/layout/activity_main.xml
	app/src/main/res/layout/activity_meas_heart_rate.xml
	app/src/main/res/layout/layout.xml
	app/src/main/res/values/colors.xml
	app/src/main/res/values/dimens.xml
	app/src/main/res/values/strings.xml
	data/build.gradle
	data/src/main/java/com/example/data/database/Converters.kt
	data/src/main/java/com/example/data/database/dao/HeartDataDao.kt
	data/src/main/java/com/example/data/repository/HeartRateMeasureRepository.kt
	domain/src/main/java/com/example/domain/di/InteractionModule.kt
	domain/src/main/java/com/example/domain/repository/IHeartRateHistoryRepository.kt

akshaya86 committed:
HRM04: Merge conflict resolved
